File: composeApp/src/androidMain/kotlin/org/gce/racehub/profile/ProfileState.kt
package org.gce.racehub.profile

import org.gce.racehub.auth.domain.model.User
import org.gce.racehub.race.domain.model.UserProfile

data class ProfileState(
    val user: User? = null,
    val profile: UserProfile? = null,
    val isLoadingProfile: Boolean = false,
    val isSigningOut: Boolean = false,
    val errorMessage: String? = null
)

File: iosApp/iosApp/Profile/ProfileState.swift
import Foundation
import Shared

struct ProfileState {
    var user: User? = nil
    var isLoadingProfile: Bool = false
    var postsCount: Int = 0
    var savedCount: Int = 0
    var recentThreadTitles: [String] = []
    var savedThreadTitles: [String] = []
    var isSigningOut: Bool = false
    var errorMessage: String? = nil
}
